### PR TITLE
Adding transform for scoring TorchScript models

### PR DESCRIPTION
**Pull request by interesaaat**
This PR adds a TorchTransform allowing to score TorchScript models in ML.NET. This addition requires LibTorchSharp and TorchSharp dlls.

### CODE REVIEW

File: src/Microsoft.ML.Torch/TorchModel.cs [hunk 1]
```diff
@@ -32,14 +33,32 @@ internal TorchModel(IHostEnvironment env, TorchSharp.JIT.Module module, string m
...
+        // I guess it's a utility method to inspect the schema of the model when you load the model. Is it the input schema or the output schema?
+        public DataViewSchema GetModelSchema()
+        {
+            throw new NotImplementedException();
+        }
+
         /// <summary>
-        /// Get <see cref="DataViewSchema"/> for complete model.
-        /// Every node in the Torch model will be included in the <see cref="DataViewSchema"/> object.
+        /// Scores a dataset using a pre-traiend <a href="https://www.pytorch.org/">Torch</a> model.
```
Comment (interesaaat) [inline src/Microsoft.ML.Torch/TorchModel.cs:44]: I think we should say that this is for TorchScript models only.

File: src/Microsoft.ML.Torch/TorchScoringTransformer.cs [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+
+                        for (int j = 0; j < colShapeLength; j++)
+                        {
+                            if (colTypeDims[j] != _parent.InputShapes[i][j])
+                                throw Host.Except($"Input shape mismatch: Input Column '{_parent.InputColumnNames[i]}' dimension {j} of size {colTypeDims[j]} does not match expected size {_parent.InputShapes[j]}.");
+                        }
+                    }
+                    else
+                    {
+                        // If the column in the input schema is one dimension we make sure that the total size of the TF shape matches.
```
Comment (interesaaat) [inline src/Microsoft.ML.Torch/TorchScoringTransformer.cs]: Torch instead of TF ;)

File: src/Microsoft.ML.Torch/TorchScoringTransformer.cs [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+                    var resultSize = result.NumberOfElements;
+                    var editor = VBufferEditor.Create(ref dst, (int)resultSize);
+
+                    result.Data.CopyTo(editor.Values);
+                    dst = editor.Commit();
+                    result.Dispose();
+                };
+
+                // REVIEW: not sure we need the below. Alos, why are we not disposing the input tensors?
+                disposer = () =>
```
Comment (interesaaat) [inline src/Microsoft.ML.Torch/TorchScoringTransformer.cs]: We actually have to I believe. This is related to the struct vs class discussion we had. I think that calling dispose explicitly is safer.

File: test/Microsoft.ML.Tests/Scenarios/TorchTests.cs [hunk 4]
```diff
@@ -37,12 +41,37 @@ public void TorchMNISTScoringTest()
...
+             var count = mlContext.Data.CreateEnumerable<MINSTOutputData>(output, false).ToArray()[0].TorchOutput.Length;
+            Assert.True(count == 1000);
+        }
+
+        [TorchFact]
+        public void TorchMNISTWorkoutTest()
+        {
+            var mlContext = new MLContext();
+            var ones = FloatTensor.Ones(new long[] { 1, 3, 224, 224 });
+            var datapoint = new MINSTInputData
```
Comment (interesaaat) [inline test/Microsoft.ML.Tests/Scenarios/TorchTests.cs]: MINST => MNIST. I introduced this typo :)

File: src/Microsoft.ML.Torch/TorchScoringTransformer.cs (hunk 2 above)
Comment (artidoro) [reply src/Microsoft.ML.Torch/TorchScoringTransformer.cs]: Thank you!

File: src/Microsoft.ML.Torch/TorchScoringTransformer.cs (hunk 3 above)
Comment (artidoro) [reply src/Microsoft.ML.Torch/TorchScoringTransformer.cs]: Ok let me know if this looks any better.

File: src/Microsoft.ML.Torch/TorchModel.cs (hunk 1 above)
Comment (artidoro) [reply src/Microsoft.ML.Torch/TorchModel.cs:44]: I missed this, sorry! But I think we definitely need to update the documentation.

**Pull request merged**